catalog-scripts/idp-catalog-wizard.py

```diff
@@ -0,0 +1,174 @@
+            session.mount("https://", HTTPAdapter(max_retries=retries))
+
+            try:
+                api_response = session.post(api_url, json=api_payload, headers=api_headers)
+                if api_response.status_code == 200 or api_response.status_code == 201:
+                    print(f"Location registered for file: {repo_name}")
+                    count += 1
+                elif api_response.status_code == 409:
+                    refresh_payload = {
+                        "entityRef":f"component:default/{repo_name}"
+                    }
+                    refresh_url = f"https://idp.harness.io/{account}/idp/api/catalog/refresh"
+                    api_response = session.post(refresh_url, json=refresh_payload, headers=api_headers)
+                    print(f"Location already exists for file: {repo_name}. Refreshing it")
+                    count += 1
+                else:
+                    print(f"Failed to register location for file: {repo_name}. Status code: {api_response.status_code}")
+            except requests.exceptions.RequestException as e:
+                print(f"Failed to make API call for file: {repo_name}. Error: {str(e)}")
+
+def push_yamls():
+    print("Pushing YAMLs...")
+    subprocess.run(["git", "add", "services/"])
+    commit_message = "Adding YAMLs"
+    subprocess.run(["git", "commit", "-m", commit_message])
+    subprocess.run(["git", "push", "-f"])
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="List repositories in a GitHub organization and manage catalog-info.yaml files")
+    parser.add_argument("--org", help="GitHub organization name")
+    parser.add_argument("--token", help="GitHub personal access token")
+    parser.add_argument("--repo-pattern", help="Optional regex pattern to filter repositories")
+    parser.add_argument("--create-yamls", action="store_true", help="Create or update catalog-info.yaml files")
+    parser.add_argument("--register-yamls", action="store_true", help="Register existing catalog-info.yaml files")
+    parser.add_argument("--run-all", action="store_true", help="Run all operations: create, register, and run")
+    parser.add_argument("--x_api_key", help="Harness x-api-key")
+    parser.add_argument("--account", help="Harness account")
+    return parser.parse_args()
+
+def main():
+    args = parse_arguments()
+    
+    if not (args.create_yamls or args.register_yamls or args.run_all):
+        print("Error: One of --create-yamls, --register-yamls or --run_all must be used.")
+        return
+    
+    if args.create_yamls:
+        if args.org == None or args.token == None:
+            print("Please provide org name and token to create yamls")
+            exit()
+        list_repositories(args.org, args.token, args.repo_pattern)
+    elif args.register_yamls:
+        if args.x_api_key == None or args.account == None:
+            print("Please provide org name, token, account and x_api_key to create yamls")
```

Comment:
```suggestion
            print("Please provide GitHub org name, GitHub token, Harness account ID and Harness x_api_key to create the YAMLs")
```

Reply:
though i don't think we are using the github token in this case

Reply:
Token for auth

Reply:
I don't see the token being passed in register_yamls()